pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/readers/PinotSegmentRecordReader.java

```diff
@@ -132,6 +133,15 @@ public void init(MutableSegment mutableSegment, @Nullable int[] sortedDocIds) {
     init(mutableSegment, false, null, sortedDocIds, null, false);
   }
 
+  /**
+   * Initializes the record reader from a immutable segment.
```

Comment:
Move this function in front of the mutable segment one
```suggestion
   * Initializes the record reader from an immutable segment.
```